proposals/lip-0052.md

```diff
@@ -2295,7 +2295,80 @@ Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets fo
 
 ### Endpoints for Off-Chain Services
 
-This section specifies the non-trivial or recommended endpoints of the NFT module and does not include all endpoints. TBA
+This section specifies the non-trivial or recommended endpoints of the NFT module and does not include all endpoints. 
+
+#### hasNFT
+
+This function checks whether an NFT exists and is owned by a specific user.
+
+```python
+def hasNFT(address: Address, nftID: NFTID) -> bool:
+    if NFTStore[nftID] does not exist:
+        raise Exception("NFT substore entry does not exist")
+    return NFTStore[nftID].owner == address
+```
+
+#### getNFT
+
+This function checks, for a provided NFT ID, whether an NFT exists and returns NFT and user substores information.
+
+```python
+def getNFT(nftID: NFTID) -> dict:
+    if NFTStore[nftID] does not exist:
+        raise Exception("NFT substore entry does not exist")
+    owner = NFTStore[nftID].owner    
+    attributesArray = [
+            {
+                "module": attribute.module,
+                "attributes": attribute.attributes
+            }
+            for attribute in NFTStore[nftID].attributesArray
+    ]
+    nftIDInfo = {
+        "owner": owner,
+        "attributesArray": attributesArray
+    }
+    if len(owner) == LENGTH_ADDRESS:
```

Comment:
I see now that this check is not in the SDK (see [here](https://github.com/LiskHQ/lisk-sdk/blob/883430a25a1f0744892fb275854f29aa1af42209/framework/src/modules/nft/endpoint.ts#L116)). But from what I can see the check is necessary for the case of escrowed NFTs. I think you should let the SDK know that the endpoint has a mistake and should be updated.

Reply:
Yes, I was aware of that miss by SDK, but it seemed minor. On the other look, it is not hard to incorporate it in the SDK, so I will notify them about it.

Reply:
As pointed out by @has5aan, this issue is already resolved in SDK (see [here](https://github.com/LiskHQ/lisk-sdk/pull/8716)), so I am resolving this conversation.